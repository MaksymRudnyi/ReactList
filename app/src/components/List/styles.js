import styled from 'styled-components';

const Container = styled.div`
  max-width: 1024px;
  margin: 0 auto;
  padding: 15px;
`;
Container.displayName = 'Container';

export {
	Container
};
