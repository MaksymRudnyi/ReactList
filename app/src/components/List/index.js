import React from 'react';
import PropTypes from 'prop-types';

import Item from '../Item';
import { Container } from './styles';

const List = ({ list }) => (
  <Container>
    {list.length
      ? list.map(item => <Item key={item.id} {...item} />)
      : <h2>Sorry, there are no items.</h2>}
    {}
  </Container>
);

export default List;

List.propTypes = {
  list: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      name: PropTypes.string.isRequired,
      logo: PropTypes.string,
      city: PropTypes.string,
      speciality: PropTypes.arrayOf(PropTypes.string),
    }),
  ).isRequired,
};
