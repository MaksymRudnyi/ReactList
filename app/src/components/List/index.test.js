import React from 'react';
import toJson from 'enzyme-to-json';

import { shallow, mount } from 'enzyme';

import List from './';

describe('List', () => {
	let sut;
	let props;

	const data = [
		{
			id: 1,
			name: 'Company one',
			logo: '',
			city: 'Berlin',
			speciality: ['Excavations', 'Plumbing', 'Electrical'],
		},
		{
			id: 2,
			name: 'Company two',
			logo: '',
			city: 'Munich',
			speciality: ['Excavations', 'Electrical'],
		},
	];

	props = {
		list: data,
	};

	it('should match snapshot', () => {
		sut = shallow(<List {...props} />);

		expect(toJson(sut)).toMatchSnapshot();
	});

	it('should match snapshot with empty data', () => {
		props = {
			list: [],
		};

		sut = shallow(<List {...props} />);

		expect(toJson(sut)).toMatchSnapshot();
	});
});
