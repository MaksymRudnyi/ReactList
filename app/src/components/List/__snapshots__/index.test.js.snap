// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should match snapshot 1`] = `
<Container>
  <Item
    city="Berlin"
    id={1}
    key="1"
    logo=""
    name="Company one"
    speciality={
      Array [
        "Excavations",
        "Plumbing",
        "Electrical",
      ]
    }
  />
  <Item
    city="Munich"
    id={2}
    key="2"
    logo=""
    name="Company two"
    speciality={
      Array [
        "Excavations",
        "Electrical",
      ]
    }
  />
</Container>
`;

exports[`List should match snapshot with empty data 1`] = `
<Container>
  <h2>
    Sorry, there are no items.
  </h2>
</Container>
`;
